src/system_data.jl

```diff
@@ -378,6 +398,8 @@ function transform_single_time_series!(
                 interval,
             )
             check_add_time_series!(data.time_series_params, params)
+            !_is_uninitialized(data.time_series_params.forecast_params) &&
+                remove_time_series!(data, DeterministicSingleTimeSeries)
```

Comment:
You could reach this point if the system has Probabilistic but no DeterministicSingleTimeSeries. Your code will handle it just fine. I suggest adding a test that calls `remove_time_series!` when there are none of that type present. Someone in the future might decide to throw an exception in that case.

Reply:
This code seems to have introduced a bug. `!_is_uninitialized(data.time_series_params.forecast_params)` always returns true upon the first call to transform, as a result, remove is always called with the first transformation. Thus removing the forecast parameters just added in the previous call.